### PR TITLE
move releasenote check to the linter namespace

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Move the release note check to the linter namespace and in github workflow

### Motivation
Allow restarting the jobs by addition of the `changelog/no-changelog` label when release note is not required for instance

### CODE REVIEW

File: .github/workflows/label-analysis.yml [hunk 1]
```diff
@@ -14,18 +14,31 @@ env:
   GH_REPO: ${{ github.repository }}
 
 jobs:
-  assign-team-label:
+  assign-team-and-release-note-check:
```
Comment (KSerrania) [inline .github/workflows/label-analysis.yml:17]: Coupling both of these is weird - we're coupling an action (assign-team) to a check (release-note), which have nothing in common.

File: .github/workflows/label-analysis.yml [hunk 2]
```diff
@@ -37,7 +50,7 @@ jobs:
           labels="$(gh pr view '${{ github.event.pull_request.number }}' --json labels --jq '[.labels[].name] | (join(" "))')"
           echo "Fetched labels for PR ${{github.event.number}}: $labels"
           echo "LABELS=$labels" >> "$GITHUB_OUTPUT"
-  team-label:
+  team-label-check:
```
Comment (KSerrania) [inline .github/workflows/label-analysis.yml:53]: This check is required, renaming means requiring everyone to rebase. Do we need to do this rename?

File: .github/workflows/label-analysis.yml [hunk 3]
```diff
@@ -58,7 +71,7 @@ jobs:
           exit 1
         env:
           LABELS: ${{ needs.fetch-labels.outputs.LABELS}}
-  skip-qa:
+  skip-qa-check:
```
Comment (KSerrania) [inline .github/workflows/label-analysis.yml:74]: This check is required, renaming means requiring everyone to rebase. Do we need to do this rename.

File: .github/workflows/label-analysis.yml (hunk 1 above)
Comment (chouetz) [reply .github/workflows/label-analysis.yml:17]: I agree, I wanted not to run in another runner as they need the same context, but this is probably an unnecessary saving

File: .github/workflows/label-analysis.yml (hunk 2 above)
Comment (chouetz) [reply .github/workflows/label-analysis.yml:53]: As I will need to change the required checks because of my move I thought it could be nice to improve these names as well, even if I agree this is not critical

File: .github/workflows/label-analysis.yml [hunk 4]
```diff
@@ -20,10 +20,37 @@ jobs:
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
-      - name: Install Python dependencies
+      - name: Setup python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
```
Comment (FlorentClarret) [inline .github/workflows/label-analysis.yml:26]: nit for another PR: would it make sense to have a variable for the python version? We currently copy/paste it everywhere

File: .github/workflows/label-analysis.yml [hunk 5]
```diff
@@ -20,10 +20,37 @@ jobs:
...
       - name: Auto assign team label
         run: inv -e github.assign-team-label --pr-id='${{ github.event.pull_request.number }}'
+  release-note-check:
+    if: github.triggering_actor != 'dd-devflow[bot]'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```
Comment (FlorentClarret) [inline .github/workflows/label-analysis.yml:40]: ❓ question: Why do we need to get the entire history? It looks like we only have a look at the labels and if the PR contains a releasenote file

File: tasks/requirements_release_tasks.txt [hunk 6]
```diff
@@ -1 +1,2 @@
-atlassian-python-api==3.41.3
\ No newline at end of file
+atlassian-python-api==3.41.3
+reno==3.5.0
```
Comment (FlorentClarret) [inline tasks/requirements_release_tasks.txt:2]: ❓ question: How do we decide the version lof reno we use? This version is already 3 years old. Would it make sense to bump it in another PR?

**Review by FlorentClarret — COMMENTED** [review]
I left some questions but looks great!

File: .github/workflows/label-analysis.yml (hunk 5 above)
Comment (chouetz) [reply .github/workflows/label-analysis.yml:40]: It's because of `reno`. It looks all commits until the last tag, and I don't know if I can specify this (get history until the last tag) easily

File: tasks/requirements_release_tasks.txt (hunk 6 above)
Comment (chouetz) [reply tasks/requirements_release_tasks.txt:2]: Yes I'm sure it would make sense. I would certainly encourage us to benefit from a dependabot thing to stay up to date

File: .github/workflows/label-analysis.yml (hunk 5 above)
Comment (chouetz) [reply .github/workflows/label-analysis.yml:40]: I think we can reach something nice using
```
git clone --filter=blob:none --sparse git@github.com:DataDog/datadog-agent.git
git sparse-checkout add releasenotes releasenotes/dca
```
That would do a lazy checkout of only requested files. I'll test this on another PR if you agree.

**Review by FlorentClarret — APPROVED** [review]

File: tasks/linter.py [hunk 7]
```diff
@@ -383,3 +385,29 @@ def gitlab_ci(_, test="all", custom_context=None):
...
             raise Exit(code=1)
+
+
+@task
+def releasenote(ctx):
+    """
+    Lint release notes with Reno
+    """
+    branch = os.environ.get("BRANCH_NAME")
+    pr_url = os.environ.get("PR_ID")
```
Comment (KevinFairise2) [inline tasks/linter.py:396]: That's a bit strange to map the env var `PR_ID` to `pr_url`.
Reading the following it looks like the `PR_ID` variables refert to `PR_URL`
```suggestion
    pr_url = os.environ.get("PR_URL")
```

Comment (chouetz) [reply tasks/linter.py:396]: Didn't you recognise a mutation test 😄
Thanks for the check, I think it just worked because the URL is only requested to get the ID, so maybe the ID was not accessible directly on circleci

File: tasks/linter.py [hunk 8]
```diff
@@ -393,3 +395,28 @@ def gitlab_ci(_, test="all", custom_context=None):
...
+                print(
+                    "Error: No releasenote was found for this PR. Please add one using 'reno'"
+                    ", see https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno"
+                    ", or apply the label 'changelog/no-changelog' to the PR."
+                )
+                raise Exit(code=1)
+        else:
+            print("'changelog/no-changelog' label found on the PR: skipping linting")
+
+    ctx.run("reno lint")
```
Comment (CelianR) [inline tasks/linter.py:422]: Should this be part of the `if` block ?

File: tasks/linter.py [hunk 9]
```diff
@@ -393,3 +395,28 @@ def gitlab_ci(_, test="all", custom_context=None):
...
+    run_check = is_pr_context(branch, pr_id, "release note")
+    if run_check:
+        github = GithubAPI(repository=GITHUB_REPO_NAME, public_repo=True)
+        if github.is_release_note_needed(pr_id):
+            if not github.contains_release_note(pr_id):
+                print(
+                    "Error: No releasenote was found for this PR. Please add one using 'reno'"
+                    ", see https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno"
+                    ", or apply the label 'changelog/no-changelog' to the PR."
+                )
```
Comment (CelianR) [inline tasks/linter.py:417]: **🥜 nitpick:** ‏We can display it on stderr / in red

**Review by CelianR — APPROVED** [review]
Some NITs, lgtm otherwise
